tests/unit-tests/test-subscription-integration.php

```diff
@@ -0,0 +1,432 @@
+				],
+			]
+		);
+
+		add_user_meta(
+			self::$user_with_cancelled_sub_on_2,
+			Subscription_Changed::USER_SUBSCRIPTIONS_META_KEY,
+			[
+				'http://site2' => [
+					502 => [
+						'id'       => 502,
+						'name'     => 'Subscription 502',
+						'status'   => 'cancelled',
+						'products' => [
+							220 => [
+								'id'   => 220,
+								'name' => 'Plan 220',
+							],
+						],
+					],
+				],
+			]
+		);
+
+		add_user_meta(
+			self::$user_with_non_network_sub_on_1,
+			Subscription_Changed::USER_SUBSCRIPTIONS_META_KEY,
+			[
+				'http://site1' => [
+					503 => [
+						'id'       => 503,
+						'name'     => 'Subscription 503',
+						'status'   => 'active',
+						'products' => [
+							999 => [
+								'id'   => 999,
+								'name' => 'Plan 999',
+							],
+						],
+					],
+				],
+			]
+		);
+	}
+
+	/**
+	 * Test get_network_membership_plans
+	 */
+	public function test_get_network_membership_plans() {
+		$method = self::get_method( 'get_network_membership_plans' );
+
+		$result = $method->invoke( null, 'net-1' );
+		$this->assertCount( 2, $result );
+		$this->assertArrayHasKey( 'http://site1', $result );
+		$this->assertArrayHasKey( 'http://site2', $result );
+		$this->assertArrayHasKey( 1, $result['http://site1'] );
+		$this->assertArrayHasKey( 1, $result['http://site2'] );
+
+		$result = $method->invoke( null, 'net-2' );
+		$this->assertCount( 2, $result );
+		$this->assertArrayHasKey( 'http://site1', $result );
+		$this->assertArrayHasKey( 'http://site2', $result );
+		$this->assertArrayHasKey( 2, $result['http://site1'] );
+		$this->assertArrayHasKey( 20, $result['http://site2'] );
+
+		$result = $method->invoke( null, 'net-3' );
+		$this->assertEquals( [], $result );
+	}
+
+	/**
+	 * Test get_user_network_active_subscriptions
+	 */
+	public function test_get_user_network_active_subscriptions() {
+		$method = self::get_method( 'get_user_network_active_subscriptions' );
+
+		$result = $method->invoke( null, self::$user_with_sub_on_1 );
+		$this->assertCount( 1, $result );
+		$this->assertArrayHasKey( 'http://site1', $result );
+		$this->assertCount( 1, $result['http://site1'] );
+		$this->assertArrayHasKey( 500, $result['http://site1'] );
+
+		$result = $method->invoke( null, self::$user_with_sub_on_2 );
+		$this->assertCount( 1, $result );
+		$this->assertArrayHasKey( 'http://site2', $result );
+		$this->assertCount( 1, $result['http://site2'] );
+		$this->assertArrayHasKey( 501, $result['http://site2'] );
+
+		$result = $method->invoke( null, self::$user_with_cancelled_sub_on_2 );
+		$this->assertEquals( [], $result );
+
+		$result = $method->invoke( null, self::$user_with_non_network_sub_on_1 );
+		$this->assertCount( 1, $result );
+		$this->assertArrayHasKey( 'http://site1', $result );
+		$this->assertCount( 1, $result['http://site1'] );
+		$this->assertArrayHasKey( 503, $result['http://site1'] );
+
+		$result = $method->invoke( null, self::$user_without_subs );
+		$this->assertEquals( [], $result );
```

Comment:
These tests are a bit hard to read with the generic variable names. It'd be more readable if `$method` was called e.g. `$get_user_network_active_subscriptions_method` and `$result` e.g. `$user_network_active_subscriptions`.

Reply:
https://github.com/Automattic/newspack-network/pull/125/commits/682f694368e9f1471de184e06fdbf822bd6b1a07